provisioning/Samples/service/EnrollmentGroupSample/Program.cs

```
@@ -1,34 +1,44 @@
-
-        // For this sample either:
-        // - pass this value as a command-prompt argument
-        // - set the PROVISIONING_CONNECTION_STRING environment variable 
-        // - create a launchSettings.json (see launchSettings.json.template) containing the variable
+        /// <summary>
+        /// A sample to manage enrollment groups in device provisioning service.
+        /// </summary>
+        /// <param name="args">
+        /// Run with `--help` to see a list of required and optional parameters.
+        /// </param>
         private static string s_connectionString = Environment.GetEnvironmentVariable("PROVISIONING_CONNECTION_STRING");
 
         public static int Main(string[] args)
         {
-            if (args.Length < 1)
-            {
-                Console.WriteLine("EnrollmentGroupSample <groupIssuerCertificate.cer>");
-                return 1;
-            }
+            // Parse application parameters
+            Parameters parameters = null;
+            ParserResult<Parameters> result = Parser.Default.ParseArguments<Parameters>(args)
+                .WithParsed(parsedParams =>
+                {
+                    parameters = parsedParams;
+                })
+                .WithNotParsed(errors =>
+                {
+                    Environment.Exit(1);
+                });
 
-            X509Certificate2 certificate = new X509Certificate2(args[0]);
+            X509Certificate2 certificate = new X509Certificate2(parameters.CertificatePath);
 
-            if (string.IsNullOrEmpty(s_connectionString) && args.Length > 1)
+            // The ProvisioningConnectionString argument is not required when either:
+            // - set the PROVISIONING_CONNECTION_STRING environment variable 
+            // - create a launchSettings.json (see launchSettings.json.template) containing the variable
+            if (string.IsNullOrEmpty(s_connectionString) && !string.IsNullOrEmpty(parameters.ProvisioningConnectionString))

```

seems like if neither of them is present, we should throw an argument exception here.

---

> seems like if neither of them is present, we should throw an argument exception here.

modified codes by referring to [this](https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/main/provisioning/Samples/device/TpmSample/Program.cs#L40)